superduper/backends/ibis/data_backend.py

```diff
@@ -131,6 +131,29 @@ def insert(self, table_name, raw_documents):
             if self.conn.backend_table_type == DataFrame:
                 df.to_csv(os.path.join(self.name, table_name + '.csv'), index=False)
 
+    def check_ready_ids(
+        self, query: IbisQuery, keys: t.List[str], ids: t.Optional[t.List[t.Any]] = None
+    ):
+        """Check if all the keys are ready in the ids.
+
+        :param query: The query object.
+        :param keys: The keys to check.
+        :param ids: The ids to check.
+        """
+        if ids:
+            query = query.filter(query[query.primary_id].isin(ids))
+        conditions = []
+        for key in keys:
+            conditions.append(query[key].notnull())
+        docs = query.filter(*conditions).select(query.primary_id).execute()
+        ready_ids = [doc[query.primary_id] for doc in docs]
+        if ids and len(ready_ids) != len(ids):
```

Comment:
we should use the log messages and calculation of not_ready_ids from base 
so that we maintain same messages in mongodb and Ibis

Reply:
Done